src/struct_utils.cc

```diff
@@ -102,6 +103,37 @@ StructUtils::FindResult StructUtils::GetStringList(
   return WRONG_TYPE;
 }
 
+StructUtils::FindResult StructUtils::GetPrimitiveFieldToString(
+    const std::string& name, std::string* out_string) {
+  const ::google::protobuf::Value* found;
+  FindResult result = FindNestedField(name, found);
+  if (result != OK) {
+    return result;
+  }
+  if (found->kind_case() == google::protobuf::Value::kNumberValue) {
+    if (found->number_value() < 0 ||
+        found->number_value() >=
+            static_cast<double>(std::numeric_limits<uint64_t>::max())) {
+      return OUT_OF_RANGE;
+    }
+    *out_string = std::to_string(static_cast<uint64_t>(found->number_value()));
```

Comment:
don't cast it to uint64_t,  just cast from double directly.   This function could be used by others too.

Reply:
If you have your own format to convert,  I suggest you could add a new Get function,  GetValue() with the same parameters as FindNestField.  Or just rename FindNestedField to GetValue(), move it to public

Reply:
Actually,  I prefer the rename approach, so that you can do any format conversion as you like in your code, not here